src/instruments/src/Common/persistence.tsx

```diff
@@ -59,18 +15,20 @@ export const usePersistentProperty = (propertyName: string, defaultValue?: strin
     }
 
     const propertySetter = (value: string) => {
-        NXDataStore.set(propertyName, value);
+        NXDataStore.set<string>(propertyName, value);
         rawPropertySetter(value);
     };
 
     return [propertyValue, propertySetter];
 };
 
 export const usePersistentPropertyWithDefault = (propertyName: string, defaultValue: string): [string, (string) => void] => {
-    const [propertyValue, rawPropertySetter] = useState(() => NXDataStore.get(propertyName, defaultValue));
+    const [propertyValue, rawPropertySetter] = useState(() => NXDataStore.get<string>(propertyName, defaultValue));
+
+    NXDataStore.subscribe<string>(propertyName, (key, value) => rawPropertySetter(value));
```

Comment:
**issue:** This callback is never cancelled. A `useEffect` hook should be used to remove the subscription.

See: https://reactjs.org/docs/hooks-reference.html#cleaning-up-an-effect